src/jl_uv.c

```diff
@@ -59,10 +59,10 @@ void JL_UV_LOCK(void)
     if (jl_mutex_trylock(&jl_uv_mutex)) {
     }
     else {
-        jl_atomic_fetch_add(&jl_uv_n_waiters, 1);
+        jl_atomic_fetch_add_relaxed(&jl_uv_n_waiters, 1);
         jl_wake_libuv();
         JL_LOCK(&jl_uv_mutex);
-        jl_atomic_fetch_add(&jl_uv_n_waiters, -1);
+        jl_atomic_fetch_add_relaxed(&jl_uv_n_waiters, -1);
```

Comment:
getting the lock ends up being acq_rel, because of the cmpswap

but F&I is also frequently seq-cst on current processors, so perhaps we should just accept that?

Reply:
It is also supposed to form a happens-before pair with the jl_wake_libuv call, not the lock

(the second call to F&D is unimportant, since we have a sequenced-before edge there anyways)

Reply:
Yeah, I think it's fine as-is.

> but F&I is also frequently seq-cst on current processors, so perhaps we should just accept that?

I played around with different processors a bit on godbolt. It looks like F&I produces slightly different code for seq_cst and relaxed on ARMv8 and Power: https://godbolt.org/z/bnvrYvfz4 I don't know about the performance implication of this, though.

Reply:
I like to keep this processor reference handy also: https://www.cl.cam.ac.uk/~pes20/cpp/cpp0xmappings.html

You are right, that this may be more expensive, so if we don't need it, it is better to specify exactly what we do need.